oauth2_http/javatests/com/google/auth/oauth2/ServiceAccountCredentialsTest.java

```diff
@@ -364,6 +369,81 @@ public void serialize() throws IOException, ClassNotFoundException {
     assertSame(deserializedCredentials.clock, Clock.SYSTEM);
   }
 
+  @Test
+  public void fromStream_nullTransport_throws() throws IOException {
+    InputStream stream = new ByteArrayInputStream("foo".getBytes());
+    try {
+      ServiceAccountCredentials.fromStream(stream, null);
+      fail();
+    } catch (NullPointerException expected) {
+      // Expected
+    }
+  }
+
+  @Test
+  public void fromStream_nullStreamThrows() throws IOException {
+    MockHttpTransportFactory transportFactory = new MockHttpTransportFactory();
+    try {
+      ServiceAccountCredentials.fromStream(null, transportFactory);
+      fail();
+    } catch (NullPointerException expected) {
+      // Expected
+    }
+  }
+
+  @Test
+  public void fromStream_providesToken() throws IOException {
+    MockTokenServerTransportFactory transportFactory = new MockTokenServerTransportFactory();
+    transportFactory.transport.addServiceAccount(SA_CLIENT_EMAIL, ACCESS_TOKEN);
+    InputStream serviceAccountStream = ServiceAccountCredentialsTest
+        .writeServiceAccountAccountStream(
```

Comment:
1. It seems like this method's name is wrong (two "Account"s in it)
2. You don't need to qualify it since it's within this class, right?